ampd/src/stellar/verifier.rs

```diff
@@ -372,19 +353,23 @@ mod test {
             },
         };
 
-        let weighted_signers: ScVal =
+        let weighted_signers_hash = BytesM::try_from(
             WeightedSigners::try_from(&verifier_set_confirmation.verifier_set)
                 .unwrap()
-                .try_into()
-                .unwrap();
+                .hash()
+                .unwrap(),
+        )
+        .unwrap();
 
         let event_body = ContractEventBody::V0(ContractEventV0 {
-            topics: vec![ScVal::Symbol(ScSymbol(
-                StringM::from_str(TOPIC_ROTATED).unwrap(),
-            ))]
+            topics: vec![
+                ScVal::Symbol(ScSymbol(StringM::from_str(TOPIC_ROTATED).unwrap())),
+                ScVal::U64(1),
+                ScVal::Bytes(ScBytes(weighted_signers_hash)),
```

Comment:
```suggestion
                ScVal::Bytes(ScBytes(BytesM::try_from(weighted_signers_hash).unwrap())),
```
convert it here perhaps for consistency with other topic values above

Reply:
Agreed. updated.

Reply:
after updating the code, it gives lint error:
```
error: useless conversion to the same type: `stellar_xdr::curr::BytesM<u32::MAX>`
   --> ampd/src/stellar/verifier.rs:368:38
    |
368 |                 ScVal::Bytes(ScBytes(BytesM::try_from(weighted_signers_hash).unwrap())),
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: consider removing `BytesM::try_from()`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
    = note: `-D clippy::useless-conversion` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::useless_conversion)]`

error: could not compile `ampd` (lib test) due to 2 previous errors
```

Thus, reverted

Reply:
well, you need to remove the BytesM conversion where the variable is being created first since it's being done here